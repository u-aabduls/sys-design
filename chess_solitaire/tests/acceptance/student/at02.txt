-- ACCEPTANCE TEST #2: SUCCESS CASE: EASY

-- Setup some chess pieces for the game
setup_chess (N,1,3)
setup_chess (B,2,2)
setup_chess (B,3,3)
setup_chess (P,4,2)


-- Start the game
start_game

-- Show all possible moves for Knight@(1,3)
moves(1,3)

-- Show all possible moves for Bishop@(2,2)
moves(2,2)

-- Show all possible moves for Bishop@(3,3)
moves(3,3)

-- Show all possible moves for Pawn@(4,2)
moves(4,2)


-- Pawn capture Bishop@(3,3)
move_and_capture(4,2,3,3)

-- Show all possible moves for Pawn@(3,3)
moves(3,3)

-- Pawn capture Bishop@(2,2)
move_and_capture(3,3,2,2)

-- Show all possible moves for Pawn@(2,2)
moves(2,2)

-- Pawn capture Knight@(1,3)
move_and_capture(2,2,1,3)

-- Game Over: You Win!