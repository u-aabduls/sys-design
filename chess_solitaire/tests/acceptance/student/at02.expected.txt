  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,1,3)
  # of chess pieces on board: 1
  Game being Setup...
  ..N.
  ....
  ....
  ....
->setup_chess(B,2,2)
  # of chess pieces on board: 2
  Game being Setup...
  ..N.
  .B..
  ....
  ....
->setup_chess(B,3,3)
  # of chess pieces on board: 3
  Game being Setup...
  ..N.
  .B..
  ..B.
  ....
->setup_chess(P,4,2)
  # of chess pieces on board: 4
  Game being Setup...
  ..N.
  .B..
  ..B.
  .P..
->start_game
  # of chess pieces on board: 4
  Game In Progress...
  ..N.
  .B..
  ..B.
  .P..
->moves(1,3)
  # of chess pieces on board: 4
  Game In Progress...
  ..N.
  +...
  .+.+
  ....
->moves(2,2)
  # of chess pieces on board: 4
  Game In Progress...
  +.+.
  .B..
  +.+.
  ...+
->moves(3,3)
  # of chess pieces on board: 4
  Game In Progress...
  +...
  .+.+
  ..B.
  .+.+
->moves(4,2)
  # of chess pieces on board: 4
  Game In Progress...
  ....
  ....
  +.+.
  .P..
->move_and_capture(4,2,3,3)
  # of chess pieces on board: 3
  Game In Progress...
  ..N.
  .B..
  ..P.
  ....
->moves(3,3)
  # of chess pieces on board: 3
  Game In Progress...
  ....
  .+.+
  ..P.
  ....
->move_and_capture(3,3,2,2)
  # of chess pieces on board: 2
  Game In Progress...
  ..N.
  .P..
  ....
  ....
->moves(2,2)
  # of chess pieces on board: 2
  Game In Progress...
  +.+.
  .P..
  ....
  ....
->move_and_capture(2,2,1,3)
  # of chess pieces on board: 1
  Game Over: You Win!
  ..P.
  ....
  ....
  ....
